### ISSUE TITLE
Two questions: portability and post-uninstall cleanup

### ISSUE THREAD

**Issue by ltGuillaume** [ScoopInstaller/Scoop#4162]
_TLDR; Scoop's use of the term "portable" isn't portable at all, but is it meant to be? (Local)AppData folders could be symlinked to the Scoop app's persistent dir. Registry keys are trickier. When uninstalling with `--purge`, (Local)AppData and registry traces could be removed._

I'm just starting to get familiar with Scoop and from what I've seen so far, it's pretty much exactly what I'd like to start using for systems for other people (family and friends) to keep everything up-to-date and easy to manage. It certainly beats Chocolatey on the account of transparency, folder structure and package management.

Two questions do pop up, however, and searching the Wiki and the issues did not get me answers:

### Portability
What are the actual goals concerning portability? Scoop uses the term "portable" and even has a separate bucket for "non-portable" apps. As far as I can see, though, it's got nothing to do with portability, but rather with the matter of admin requirements. The term "portable" requires an app to be able to be pulled from its folder and put into another (onto another system) while its configuration is preserved (https://en.wikipedia.org/wiki/Portable_application). This is not the case with many of the apps in the main and extras bucket, as it is only the fact for those apps that are already inherently portable. There's a nice feature for persistent files, but (Local)AppData folders and registry keys seem to be completely ignored. For example, I have not (yet) seen symlinks set up from AppData folders to the app's persistent folder without Scoop in order to keep it actually contained. As for registry keys, e.g. WinMerge uses a single registry key for its settings, but Scoop totally ignores this fact, even when uninstalling the app.

### Traces after uninstalling
This is directly connected to the portability issue. Basically, none of the package jsons I've seen handle any of the traces the apps leave on the system when uninstalling. As such, registry keys, (Local)AppData folders etc. etc. are just left there. Cleaning up after itself may not be part of the hard definition of being a "portable app" (for some people at least), but it most definitely is for "uninstalling". What's the general stance about this for Scoop then?

If this is all by design, then it's rather confusing for the end user: some of it is portable, some of it isn't - even though it doesn't come from the non-portable bucket - and thus all settings will be gone if you don't back up the appropriote folders and registry keys yourself.

**Comment by Deide** [comment]
I'm no authority on Scoop, but here's my take:

Scoop _prefers_ portable applications as they're easier to install, update, remove and manage. But pretty much anything goes. I do agree that we could use some cleaning up and focusing of the buckets.

I think the main reason for those missed cases is that it's difficult to investigate and keep track of all the potential ways an application may integrate with the OS. If Winmerge is leaving registry entries behind, open an issue about it in the extras repository. Scoop can specify uninstaller scripts; though because its data is technically persistent user configuration if I understand you correctly, it should only be removed upon `-p` or `--purge`.

**Comment by ltGuillaume** [comment]
> Scoop can specify uninstaller scripts; though because its data is technically persistent user configuration if I understand you correctly, it should only be removed upon -p or --purge.

You're right, this should be taken into account. (WinMerge uses `HKEY_CURRENT_USER\Software\Thingamahoochie` to save its settings).

> I think the main reason for those missed cases is that it's difficult to investigate and keep track of all the potential ways an application may integrate with the OS.

That is certainly true: I make all the software I use portable (where possible and mostly without virtualization), and it is sometimes a pain. The main issue I've got with Scoop is however that it's not clear on the surface that it _doesn't try_ to do this by default, even though it seems that "portable" is used repeatedly in the Scoop documentation.

**Comment by ltGuillaume** [comment]
@lukesampson Any thoughts on this? 🙂

**Comment by ltGuillaume** [comment]
@rashil2000 You've closed an issue that has never been addressed or even replied to by the devs.
